xdsl/dialects/riscv.py

```diff
@@ -331,6 +333,106 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        return cls.create(
+            operands=cls.parse_operands(parser),
+            successors=cls.parse_successors(parser),
+            regions=cls.parse_regions(parser),
+            attributes=cls.parse_attributes(parser),
+            result_types=cls.parse_result_types(parser)[1],
+        )
+
+    @classmethod
+    def parse_operands(cls, parser: Parser) -> Sequence[SSAValue]:
+        operands = parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, parser.parse_operand
+        )
+        return operands
+
+    @classmethod
+    def parse_successors(cls, parser: Parser) -> Sequence[Block]:
+        return parser.parse_optional_successors() or []
```

Comment:
supernit:
```suggestion
        return parser.parse_optional_successors() or ()
```
tuples by default for Sequences seems like a good idea, but it's a minor code style thing

Reply:
I think one rule we like to follow would be to use tuples unless mutability is required, instead of defaulting to lists. Tuples are hashable by default, and have some other nice benefits over lists.

Reply:
In this case it's not all that important but they're also covariant due to the immutability